week3/api-services/address_apn_lookup.py

```
@@ -0,0 +1,68 @@
+                })
+            
+            # Return best match and alternatives
+            return {
+                "found": True,
+                "best_match": matches[0],
+                "alternatives": matches[1:] if len(matches) > 1 else []
+            }
+        else:
+            raise HTTPException(status_code=404, detail="Address not found")
+            
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)